feat(services): new services to use

linuxbandit opened the pull request:
All the service configuration has been tested on AEGEE/MyAEGEE, but of course I will try once more

linuxbandit committed:
feat(services): new services to use

serge1peshcoff commented on oms-status/docker/cachet/.env.example:
```diff
@@ -0,0 +1,45 @@
+APP_ENV=production
```
let's move this to the general .env file instead?

linuxbandit replied:
this is the PHP env, and it honestly never changes

linuxbandit replied:
because we don't have to develop this stuff

all of this stuff in PR comes out-of-the-box

linuxbandit committed:
fix(docker): put correct docker-compose version

linuxbandit committed:
feat(services): add monitoring

serge1peshcoff commented on oms-wiki/docker/LocalSettings.php.example:
```diff
@@ -0,0 +1,146 @@
+$wgEnableUserEmail = false; # UPO
+
+$wgEmergencyContact = "fabrizio@barnab.eu";
+$wgPasswordSender = "fabrizio@barnab.eu";
```
I suggest making this one and the one above non personal

linuxbandit replied:
true

linuxbandit committed:
chore(modules): merge-squash branch master in impr/new_services

linuxbandit commented:
do you know why, if i updated this branch with the latest content with master, it shows me as diff the stuff that is already on master?!

serge1peshcoff commented:
No idea, try rebasing on master and force pushing, usually helps

serge1peshcoff commented:
Also don't merge master into this branch, there's no point and it also creates the unnecessary garbage commits

linuxbandit committed:
impr: add monitor agents, add other configs

- grafana/prom improvement of the auto-provision
- service for uptime
- add helper script to configure cachet
- other minor things

linuxbandit committed:
fix: proper address

linuxbandit commented:
I don't fucking get what the fuck happened lol. I updated this branch by merging master (I could not rebase as I already published it). And now this? what the fuck?

linuxbandit committed:
fix: change targets and paths

linuxbandit committed:
impr: add slack token for having pic upload

linuxbandit committed:
impr: add provisioning of ciao

linuxbandit committed:
chore: refactor, create a folder oms-dev

linuxbandit committed:
Merge branch 'master' into impr/new_services

linuxbandit commented:
WHAT the fuck, it did ANOTHER merge master into this branch. jesus christ. Idk

linuxbandit commented:
tomorrow i will see if closing this branch and rebasing shit. now im off to bed

serge1peshcoff commented:
lol

serge1peshcoff commented on .env.example:
```diff
@@ -26,17 +45,34 @@ PATH_OMS_EVENTS=oms-events/docker/
+PATH_OMS_SHORTENER=oms-shortener/docker/
+PATH_OMS_MONITOR=oms-monitor/docker/
 PATH_OMS_ELASTIC=oms-elastic/docker/
-PATH_OMS_DISCOUNTS=oms-discounts/docker/
```
is there a point in deleting this one?

linuxbandit replied:
bad merge i think, I will add it again

serge1peshcoff commented on base-docker-compose.yml:
```diff
@@ -6,70 +6,34 @@ version: "3.4"
+        file: secrets/mail_password
     db_password:
-        file: secrets/db_password
+        file: secrets/core-db_password
```
actually a proposal: let's delete it entirely and use env vars for it? maybe not in this PR but later

serge1peshcoff replied:
and also for mail_user and mail_password, seems reasonable to use secret files for things where we really need files

linuxbandit replied:
imho for our settings, secrets and env variables are the same thing. If an attacker gets access to a container, reading a fil in /var/run/secrets (iirc that's the folder where secrets are mounted) is not much different than reading the environment variables. I am no security expert tho, so we'll have to read about stuff.

In favour about removing secrets altogether (it'd make us able to use composer version 2.4 and `limit` resources)

serge1peshcoff replied:
my concern is not the security here, but that'd be nice to standardize things here.
reasoning - for now we have some strings that are in .env and some strings that are in files, so this can be confusing. that's why I'd rather use .env for all of the strings and leave files only for, like, real files.

for using 2.4, why not moving to 3.x and use docker stack instead?

linuxbandit replied:
i told you about the docker stack, it's a complication we don't need

serge1peshcoff commented on oms-dev/docker/docker-compose.yml:
```diff
@@ -0,0 +1,131 @@
+      DEFAULT_USER: "admin@${BASE_URL}"
+      DEFAULT_PASSWORD: "${PW_PGADMIN}"
+
+### mongoui Container #######################################
```
we don't have mongodb anymore, let's delete it maybe?

linuxbandit replied:
maybe if some service needs it again, this can be used

serge1peshcoff replied:
but this would be in the history anyway, no?

linuxbandit replied:
i will remove it then
